metisp/pymetis/src/pymetis/classes/products/product.py

```diff
@@ -32,22 +34,35 @@ class PipelineProduct(ABC):
         one FITS file with associated headers and a frame
     """
 
+    # Global defaults for all Products
+    group: cpl.ui.Frame.FrameGroup = cpl.ui.Frame.FrameGroup.PRODUCT        # ToDo: Is this a sensible default?
+    level: cpl.ui.Frame.FrameLevel = None
+    frame_type: cpl.ui.Frame.FrameType = None
+
+    # Product metadata.
+    # The standard way of defining them is to override the private class attribute;
+    # the default @classmethod with the same name (without the underscore) just returns its value.
+    # If it depends on other attributes, override the corresponding @classmethod.
+    # All methods dealing with these should relate to the class, not instances!
     _tag: str = None
-    _group: cpl.ui.Frame.FrameGroup = cpl.ui.Frame.FrameGroup.PRODUCT        # ToDo: Is this a sensible default?
-    _level: cpl.ui.Frame.FrameLevel = None
-    _frame_type: cpl.ui.Frame.FrameType = None
-    _used_frames: cpl.ui.FrameSet = None
+    _oca_keywords: {str} = None
+    _description: str = None
+
+    # Use this regex to verify that the product tag is correct.
+    # This base version only verifies it is ALL_CAPS_WITH_UNDERSCORES
+    _regex_tag: re.Pattern = re.compile(r"^[A-Z0-9]+[A-Z0-9_]+[A-Z0-9]+$")
```

Comment:
Is a tag starting with a number going to work? I can imagine it leading to problems, not sure.

Reply:
I guess it is (at least I don't see a reason why it should not work). But it might be better to just forbid that.